drake/systems/plants/xmlUtil.cpp

```diff
@@ -87,13 +89,25 @@ void originAttributesToTransform(tinyxml2::XMLElement* node,
 
 void poseValueToTransform(tinyxml2::XMLElement* node, const PoseMap& pose_map,
                           Eigen::Isometry3d& T,
-                          const Eigen::Isometry3d& T_default_frame) {
+                          const Eigen::Isometry3d& T_default_frame,
+                          bool printMsgs) {
   Eigen::Vector3d rpy = Eigen::Vector3d::Zero(), xyz = Eigen::Vector3d::Zero();
   const char* strval = node->FirstChild()->Value();
   if (strval) {
     std::stringstream s(strval);
     s >> xyz(0) >> xyz(1) >> xyz(2) >> rpy(0) >> rpy(1) >> rpy(2);
   }
+
+  if (printMsgs) {
```

Comment:
> is there a style recommendation about debugging statements inside code?

Google's style guide is silent on this point as best I can tell. My 2 cents: this kind of code is great while debugging but adhoc debugging stuff should not go into the master branch, especially if there is any run time overhead (like checking a flag). With a logging/debug utility that has compile-time disabling for debug output, then I think there are times when it is worth leaving in debug output in anticipation of future needs in troublesome code.